toolkit/tools/pkg/imagecreatorlib/imagecreator.go

```diff
@@ -208,3 +218,22 @@ func createImageCreatorParameters(buildDir string,
 
 	return ic, nil
 }
+
+// getTargetOs returns the appropriate TargetOs based on distro and version
+func getTargetOs(distro string, distroVersion string) (targetos.TargetOs, error) {
```

Comment:
Would it make sense to use a nested maps instead of nested switch statements? That way, it will be more declarative.